core/db/migrate/20150113002122_create_spree_promotion_codes.rb

```diff
@@ -0,0 +1,13 @@
+class CreateSpreePromotionCodes < ActiveRecord::Migration
+  def change
+    create_table :spree_promotion_codes do |t|
+      t.references :promotion, index: true
+      t.string :value
+      t.datetime :starts_at
```

Comment:
i'm not sure i follow why codes need a start / end date. it seems that the greatest benefit (to us at least) of having multiple codes per promotion is that you can adjust their start / end date in batch by adjusting their parent promotion's start / end date, rather than having it editable on the individual code